browser/src/control/Control.Mention.ts

```diff
@@ -0,0 +1,190 @@
+		}
+	}
+
+	openMentionPopup(ev: FireEvent) {
+		var framePos = this.getCurrentCursorPosition();
+		this.users = ev.data;
+		if (this.users === null) return;
+
+		var text = this.map._docLayer._mentionText.join('').substring(1);
+		// filterout the users from list according to the text
+		if (text.length > 1) {
+			this.itemList = this.users.filter((element: any) => {
+				// case insensitive
+				return element.username.toLowerCase().includes(text.toLowerCase());
+			});
+		} else {
+			this.itemList = this.users;
+		}
+
+		if (this.itemList.length !== 0) {
+			var entries = [];
+			for (var i in this.itemList) {
+				var entry = {
+					text: this.itemList[i].username,
+					columns: [
+						{
+							text: this.itemList[i].username,
+						},
+					],
+					row: i.toString(),
+				};
+				entries.push(entry);
+			}
+
+			var data: PopupData;
+			const control = {
+				id: 'mentionList',
+				type: 'treelistbox',
+				text: '',
+				enabled: true,
+				singleclickactivate: false,
+				fireKeyEvents: true,
+				entries: [] as Array<Entry>,
+			} as MentionWidget;
+			// update the popup with list if mentionList already exist
+			if (L.DomUtil.get('mentionList')) {
+				data = {
+					jsontype: 'dialog',
+					id: 'mentionPopup',
+					action: 'update',
+					control: control,
+					posx: framePos.x,
+					posy: framePos.y,
+				} as any as PopupData;
+				(data.control as MentionWidget).entries = entries;
+				this.map.fire('jsdialogupdate', {
+					data: data,
+					callback: this.callback.bind(this),
+				});
+				return;
+			}
+			if (L.DomUtil.get('mentionPopup'))
+				this.closeMentionPopup({ typingMention: true } as CloseMessageEvent);
+			data = this.newPopupData;
+			data.children[0].children[0] = control;
+			(data.children[0].children[0] as MentionWidget).entries = entries;
+		} else {
+			const control = {
+				id: 'fixedtext',
+				type: 'fixedtext',
+				text: 'no search results found!',
+				enabled: true,
+				singleclickactivate: undefined,
+				fireKeyEvents: undefined,
+			} as MentionWidget;
+			if (L.DomUtil.get('fixedtext')) {
+				data = {
+					jsontype: 'dialog',
+					id: 'mentionPopup',
+					action: 'update',
+					control: control,
+					posx: framePos.x,
+					posy: framePos.y,
+					children: undefined,
+				} as any as PopupData;
+				this.map.fire('jsdialogupdate', {
+					data: data,
+					callback: this.callback.bind(this),
+				});
+				return;
+			}
+			if (L.DomUtil.get('mentionPopup'))
+				this.closeMentionPopup({ typingMention: true } as CloseMessageEvent);
+			data = this.newPopupData;
+			data.children[0].children[0] = control;
+		}
+		// add position
+		data.posx = framePos.x;
+		data.posy = framePos.y;
+		this.map.fire('jsdialog', {
+			data: data,
+			callback: this.callback.bind(this),
+		});
+	}
+
+	closeMentionPopup(ev: CloseMessageEvent) {
+		var closePopupData = super.getClosePopupData();
+		this.map.fire('jsdialog', { data: closePopupData, callback: undefined });
```

Comment:
these 2 lines can be shared too, it can be just single `super.sendClosePopup()` which will create and send the message